index.js

```diff
@@ -4,20 +4,76 @@
  */
 
 const got = require('got');
+const AWS = require('aws-sdk');
+
+// We use the IP address  as it is referenced from the AWS docs:
+// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html
+const baseUrl = 'http://169.254.169.254/latest';
 
 /**
- * fetch fetches the given metadata field for the currently running instance
- * and returns that data to done. If there was an error, it is passed to done
- * instead.
+ * fetch fetches the given metadata field for the currently running instance.
  * @param  {String}   field The metadata field to retrieve.
  * @param  {Promise<String>} A promise that resolves to the field data.
  */
 function fetch(field) {
-  // We use the IP address  as it is referenced from the AWS docs:
-  // https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html
-  return got(`http://169.254.169.254/latest/meta-data/${field}`).then((res) => res.text);
+  return got(`${baseUrl}/meta-data/${field}`).then((res) => res.text);
```

Comment:
Could we use the [AWS `MetadataService`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/MetadataService.html) instead?